pandas/core/dtypes/dtypes.py

```diff
@@ -1171,3 +1171,15 @@ def __from_arrow__(
             results.append(iarr)
 
         return IntervalArray._concat_same_type(results)
+
+    def _get_common_dtype(self, dtypes: List[DtypeObj]) -> Optional[DtypeObj]:
+        # NB: this doesn't handle checking for closed match
+        if not all(isinstance(x, IntervalDtype) for x in dtypes):
+            return np.dtype(object)
```

Comment:
This is the wrong return value, see docstring in the base class: https://github.com/pandas-dev/pandas/blob/36c4d5ccf1dbc02b329a55400af55de68415b7e0/pandas/core/dtypes/base.py#L326-L346

Reply:
thanks, will address in follow-up